docs/use_cases/retrieval_from_image_and_text.md

```diff
@@ -0,0 +1,227 @@
+an [RTX 3080 16 GB GPU](https://www.techpowerup.com/gpu-specs/geforce-rtx-3080.c3621), capable of 29.77 TFLOPS on FP32. 
+When processing the merged COCO training and validation dataset, containing 103,429 data samples post-preprocessing, we noted the 
+following inference times and resource allocations. It's important to note that GPU utilization was always maximized through 
+parallelized data loading to ensure efficiency.
+
+- Embedding captions with "all-mpnet-base-v2" takes approximately 5 minutes and uses about 2.8 GB of GPU memory for batches of 128.
+- Generating captions with "Salesforce/blip-image-captioning-base" spans around 3 hours and requires close to 15.5 GB of GPU memory, 
+also with batches of 128.
+- Embedding images with "tf_efficientnetv2_s.in21k_ft_in1k" similarly takes about 3 hours and consumes 15 GB of GPU memory for batch 
+sizes of 128.
+- Embedding both captions and images using the OpenCLIP "ViT-L-14_datacomp_xl_s13b_b90k" model can be completed in about 50 minutes when 
+processing with a batch size of 512, requiring 14.5 GB of GPU memory.
+
+Indeed, if high-quality image captions are already in hand, employing Sentence Transformers for embedding proves to be highly efficient. 
+This method offers a balance of speed and effectiveness. On the other hand, if only images are available and captions need to be 
+generated, the process can be time-consuming, which could be a significant factor when deciding on the method for a given application 
+or project.
+
+## Open Questions
+
+The outcomes of these experiments open up several intriguing questions for further investigation. Here are a few key areas to explore:
+
+1. The potential of other transformer-based architectures in improving results. For instance, [Data-Efficient Image Transformers](https://arxiv.org/pdf/2012.12877.pdf), 
+utilize knowledge distillation and inductive biases from ConvNets for training. Would such models offer superior performance?
+2. A closer look at [various image-captioning models](https://huggingface.co/models?other=image-captioning) to assess the quality 
+of captions they generate, particularly in relation to the size of the models. How does the caption quality vary with the complexity 
+of the models?
+3. An examination of the consistency of ConvNets' performance across diverse datasets. Do these networks consistently perform well 
+across various data environments, or does their effectiveness vary depending on the specific dataset?
+
+## Conclusion
+
+The experiments conducted in this project demonstrate that Transformer models are highly effective feature extractors for textual 
+data. State-of-the-art image-captioning models have proven to be excellent in annotating images and ensuring 
+consistency across similar concepts. ConvNets, with their inherent inductive biases, emerge as robust 
+feature encoders for image data. Moreover, the use of jointly trained text and image encoders appears to offer significant advantages 
```

Comment:
Even though what you write is true, I'd like to emphasise from the results I think what is justified is that the ViT experiments are superior to others (apart from embedding the blip generated captions). ViT encoding images only, or image + captions doesn't show that significant differences. In terms of NDCG: ViT-L-14_datacomp_xl_s13b_b90k ~50%, ViT-L-14_datacomp_xl_s13b_b90k ~53%, about 3.5%. difference. This is an important conclusion that I think misses from here, even though it is present in the Scenario 5 section, as the focus of this article is how to most efficiently encode images with textual metadata. So I'd like to see here the 3 emergent solutions in the conclusion mentioned directly: ViT Embed Image + Text, Blip captions embedded, and ViT Image only.
I would also add to the open questions that we should also test ViT (VisionTransformer) on image + embedding captions with a text model and concatenate **VS** the current best approach, ViT model on both data modalities (so basically adding a ViT based experiment to Scenario 3)

Reply:
I see your point, however I'd like to point out one thing based on the following table.

![image](https://github.com/superlinked/VectorHub/assets/10969182/009dca34-0af7-49ca-a977-ad203a84e979)

The ViT model that you mentioned is not just a ViT. It is a ViT model that was jointly trained with a Text Encoder on both images and descriptions. The simple ViT model that was trained only on images from the EmbedImage experiment is the worst performer from all, as you can see from the last row in the table.